clinical-domain-agent/src/main/java/care/smith/fts/cda/services/PatientCompartmentService.java

```diff
@@ -0,0 +1,41 @@
+package care.smith.fts.cda.services;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides information about which FHIR resource types are in the Patient compartment and which
+ * search parameters link them to a patient.
+ *
+ * <p>The compartment definition specifies for each resource type which reference fields (params)
+ * can link the resource to a patient. For example, ServiceRequest has params ["subject",
+ * "performer"], meaning it's in the patient compartment if either field references the patient.
+ */
+public class PatientCompartmentService {
```

Comment:
This service does nearly nothing. Please rename to `PatientCompartmentRepository` - move the creation of the `resourceTypeToParams` Map into a static method, incorporate the PatientCompartmentChecker functionality.